src/components/TableWrapper.tsx

```diff
@@ -13,118 +8,87 @@ import {
-}) => {
-  const [selectedRows, setSelectedRows] = useState([]);
+interface FieldData {
+  name: string;
+  data_path: string;
+  data: (string | number | boolean | null | (string | number)[])[];
+}
 
-  const toggleRow = (index) => {
-    const newSelected = selectedRows.includes(index)
-      ? selectedRows.filter((i) => i !== index)
-      : [...selectedRows, index];
-    setSelectedRows(newSelected);
+interface TableWrapperProps {
+  component: "table";
+  title: string;
+  id: string;
+  fields: FieldData[];
+  className?: string;
+}
 
-    onRowSelect?.(newSelected.map((i) => rows[i]));
-    setCustomData(newSelected.map((i) => rows[i]));
-  };
+const TableWrapper = (props: TableWrapperProps) => {
+  const { title, id, fields, className } = props;
+  // Transform fields data into table format
+  const transformFieldsToTableData = () => {
+    if (!fields || fields.length === 0) return { columns: [], rows: [] };
+
+    // Find the maximum number of data items across all fields
+    const maxDataLength = Math.max(...fields.map((field) => field.data.length));
+
+    // Create columns from field names
+    const transformedColumns = fields.map((field) => ({
+      key: field.name,
+      label: field.name,
+    }));
 
-  const toggleAllRows = () => {
-    const allSelected = selectedRows.length === rows.length;
-    const newSelected = allSelected ? [] : rows.map((_, i) => i);
-    setSelectedRows(newSelected);
+    // Create rows based on the maximum data length
+    const transformedRows = [];
+    for (let i = 0; i < maxDataLength; i++) {
+      const row: Record<string, string | number | null> = {};
+      fields.forEach((field) => {
+        const value = field.data[i];
+        if (value === null || value === undefined) {
+          row[field.name] = "N/A";
+        } else if (Array.isArray(value)) {
+          row[field.name] = value.join(", ");
+        } else {
+          row[field.name] = String(value);
+        }
+      });
+      transformedRows.push(row);
+    }
 
-    onRowSelect?.(newSelected.map((i) => rows[i]));
-    setCustomData(newSelected.map((i) => rows[i]));
+    return { columns: transformedColumns, rows: transformedRows };
   };
 
-  const graphData =
-    graph && graph.column
-      ? rows.map((row) => ({ x: row[columns[0].key], y: row[graph.column] }))
-      : [];
+  const { columns, rows } = transformFieldsToTableData();
 
   return (
-    <div>
-      <Table variant={variant} borders={variant !== "compactBorderless"}>
-        {caption && (
+    <Card id={id} className={className}>
+      <CardBody>
+        <Table variant="compact" borders>
           <Caption>
-            <div
-              style={{
-                display: "flex",
-                justifyContent: "space-between",
-                alignItems: "center",
-              }}
-            >
-              <span>{caption}</span>
-              {actions}
-            </div>
+            <span style={{ fontWeight: "bold", fontSize: "1.1em" }}>
```

Comment:
Should this stilling be here? I think CSS should be used, or some Patternfly standard class?